src/secrets.ts

```diff
@@ -0,0 +1,58 @@
+import { HDKey } from '@scure/bip32';
+import { generateMnemonic, validateMnemonic, mnemonicToSeedSync } from '@scure/bip39';
+import { wordlist } from '@scure/bip39/wordlists/english';
+import { encodeBase64toUint8 } from './base64';
+import { bytesToNumber } from './utils';
+import { hexToNumber } from '@noble/curves/abstract/utils';
+export const generateNewMnemonic = (): string => {
+	const mnemonic = generateMnemonic(wordlist, 128);
+	if (!validateMnemonic(mnemonic, wordlist)) {
+		return generateNewMnemonic();
+	}
+	return mnemonic;
+};
+
+export const deriveSeedFromMnemonic = (mnemonic: string): Uint8Array => {
+	const seed = mnemonicToSeedSync(mnemonic);
+	return seed;
+};
+
+export const deriveSecret = (seed: Uint8Array, keysetId: string, counter: number): Uint8Array => {
+	return derive(seed, keysetId, counter, 0);
+};
+
+export const deriveBlindingFactor = (
+	seed: Uint8Array,
+	keysetId: string,
+	counter: number
+): Uint8Array => {
+	return derive(seed, keysetId, counter, 1);
+};
+
+const derive = (
+	seed: Uint8Array,
+	keysetId: string,
+	counter: number,
+	secretOrBlinding: 0 | 1
+): Uint8Array => {
+	const hdkey = HDKey.fromMasterSeed(seed);
+	const keysetIdInt = getKeysetIdInt(keysetId)
+	const derivationPath = `m/129372'/0'/${keysetIdInt}'/${counter}'/${secretOrBlinding}`;
```

Comment:
How are these numbers decided? `129372'`

Reply:
in the protocol. It's the standard derivation path for cashu. We can add this path to a const to make it more clear

Reply:
Can we add a comment to the page where this derivation path is defined?